browser-extension/mv3/src/service-worker/services/messageHandler.ts

```diff
@@ -20,12 +20,16 @@ export const initMessageHandler = () => {
     /* From any case, return true when sendResponse is called asynchronously */
     switch (message.action) {
       case EXTENSION_MESSAGES.HANDSHAKE_CLIENT:
-        initClientHandler({
-          tabId: sender.tab?.id,
-          frameIds: [sender.frameId],
+        isExtensionEnabled().then((isExtensionStatusEnabled) => {
+          if (!isExtensionStatusEnabled) return;
+
+          initClientHandler({
+            tabId: sender.tab?.id,
+            frameIds: [sender.frameId],
+          });
+          applyScriptRules(sender.tab?.id, sender.frameId, sender.url);
         });
-        applyScriptRules(sender.tab?.id, sender.frameId, sender.url);
-        break;
+        return true;
```

Comment:
@nafees87n Why are we returning true from this method?

Reply:
Yeah. return true is not required here